src/ucp/stream/stream_recv.c

```diff
@@ -268,6 +283,17 @@ UCS_PROFILE_FUNC(ucs_status_ptr_t, ucp_stream_recv_nb,
     ucs_assert(req->recv.state.offset >  0);
     ucs_assert(req->recv.state.offset <= req->recv.length);
     *length = req->recv.state.offset;
+
+    if (ucs_unlikely(req != &tmp_req)) {
```

Comment:
do you really need this check? looks like check for `ucp_stream_request_can_complete(req)` is enough

Reply:
We should not call `ucp_request_put` for request allocated on stack, also we can't queue this, this is why `inplace` prediction is needed.

Reply:
right, missed that